code/game/machinery/Beacon.dm

```diff
@@ -26,7 +26,8 @@
 	Beacon.syndicate = syndicate
 	Beacon.area_bypass = area_bypass
 	Beacon.cc_beacon = cc_beacon
-	hide(T.intact)
+	if(!istype(T, /turf/simulated/floor/transparent))
```

Comment:
Pretty much all of these checks should just be a variable on turfs. 

Variables are way cleaner and more extendable than checking paths.

Reply:
Good point, ill tidy this up momentarily